### PR TITLE
DO NOT MERGE-- build hifi with neo vest plugin

### PR DESCRIPTION
**Pull request by mikeperrotta**
This PR is not yet ready to merge. I'm creating it so that I can test the code with the autobuild. Once it is in working condition, I will ask for a review.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11674/HighFidelity-Beta-PR13719-17e0fab.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11675/HighFidelity-Beta-PR13719-2f2c9d3.apk

File: plugins/neoHapticVest/src/HapticVestManager.cpp [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        SendByteArray(_touch->EncodeVibrationArray(offArray));
+    }
+}
+
+void HapticVestManager::SendByteArray(QByteArray byteArray){
+    if (serialPort) {
+        if (serialPort->isOpen()) {
+            serialPort->write(byteArray);
+        }
+        else {
```
Comment (SamGondelman) [inline plugins/neoHapticVest/src/HapticVestManager.cpp]: our coding standard has: `} else {` (else on same line as })

File: plugins/neoHapticVest/src/HapticVestManager.cpp [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    using namespace controller;
+    QVector<Input::NamedPair> availableInputs{
+    };
+    return availableInputs;
+}
+
+QString HapticVestManager::TouchDevice::getDefaultMappingConfig() const {
+    static const QString MAPPING_JSON = "";
+    return MAPPING_JSON;
+}
```
Comment (SamGondelman) [inline plugins/neoHapticVest/src/HapticVestManager.cpp]: delete

File: plugins/neoHapticVest/src/HapticVestManager.cpp [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+    if(isDifferentThanPreviousArray){
+        _parent.SendByteArray(EncodeVibrationArray(hapticInputArray));
+        for (int i = 0; i < 32; i++) {
+            previousHapticArray[i] = hapticInputArray[i];
+        }
+    }
+}
+
+void HapticVestManager::TouchDevice::focusOutEvent() {
+};
```
Comment (SamGondelman) [inline plugins/neoHapticVest/src/HapticVestManager.cpp]: delete

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11687/HighFidelity-Beta-PR13719-4d0f0f2.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11695/HighFidelity-Beta-PR13719-cb59444.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11699/HighFidelity-Beta-PR13719-dd054f0.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11704/HighFidelity-Beta-PR13719-9ef7da5.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11729/HighFidelity-Beta-PR13719-844a26c.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11733/HighFidelity-Beta-PR13719-3d2f3c0.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11742/HighFidelity-Beta-PR13719-521e9bb.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11755/HighFidelity-Beta-PR13719-20a0692.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11756/HighFidelity-Beta-PR13719-a8a0933.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11759/HighFidelity-Beta-PR13719-1189317.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11775/HighFidelity-Beta-PR13719-4a27d51.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11778/HighFidelity-Beta-PR13719-30bfa19.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11783/HighFidelity-Beta-PR13719-2edaa9e.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11783/HighFidelity-Beta-PR13719-2edaa9e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11783/HighFidelity-Beta-Interface-PR13719-2edaa9e.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11793/HighFidelity-Beta-PR13719-7b7ac90.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11793/HighFidelity-Beta-PR13719-7b7ac90.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11793/HighFidelity-Beta-Interface-PR13719-7b7ac90.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11793/HighFidelity-Beta-PR13719-7b7ac90.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11793/HighFidelity-Beta-Interface-PR13719-7b7ac90.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11812/HighFidelity-Beta-PR13719-e5e6e3f.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11812/HighFidelity-Beta-PR13719-e5e6e3f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11812/HighFidelity-Beta-Interface-PR13719-e5e6e3f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11812/HighFidelity-Beta-PR13719-e5e6e3f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11812/HighFidelity-Beta-Interface-PR13719-e5e6e3f.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12229/HighFidelity-Beta-PR13719-7006399.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12229/HighFidelity-Beta-PR13719-7006399.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12229/HighFidelity-Beta-Interface-PR13719-7006399.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12229/HighFidelity-Beta-PR13719-7006399.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12229/HighFidelity-Beta-Interface-PR13719-7006399.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12240/HighFidelity-Beta-PR13719-b434591.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12240/HighFidelity-Beta-PR13719-b434591.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12240/HighFidelity-Beta-Interface-PR13719-b434591.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12240/HighFidelity-Beta-PR13719-b434591.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12240/HighFidelity-Beta-Interface-PR13719-b434591.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12266/HighFidelity-Beta-PR13719-3e5e007.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12266/HighFidelity-Beta-PR13719-3e5e007.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12383/HighFidelity-Beta-PR13719-5d92169.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12383/HighFidelity-Beta-PR13719-5d92169.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12383/HighFidelity-Beta-Interface-PR13719-5d92169.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12383/HighFidelity-Beta-PR13719-5d92169.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12383/HighFidelity-Beta-Interface-PR13719-5d92169.exe

**Comment by hifi-gustavo** [comment]
Can someone from High Fidelity approve this PR?

**Comment by ZappoMan** [comment]
Closed for being really old/out of date.

**Pull request closed without merging**